### PR TITLE
[ROAD-823] fix: invalidate cache if solution switch

### PR DESCRIPTION
**Pull request by aldanchenko**
Fix issue with cache invalidate if solution switch to another. For this added clear OSS cache method call on different solution events.

### CODE REVIEW

**Review by michelkaporin — COMMENTED** [review]
Can we also test this, since it's a bug fix ? :-)

**Comment by aldanchenko** [comment]
@michelkaporin don't sure because this functionality related to vs sdk.

**Review by michelkaporin — APPROVED** [review]
@aldanchenko you could potentially mock the solution service and invoke event methods, then verifying if `Clean()` was called on the solution service.

Since it's probably too much for a unit test, we can leave this out, unless you find it's still worth testing. Approving for now. 

Don't forget to update changelog with this fix.

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,4 +3,5 @@
 ## [1.1.7]
 
 ### Fixed
-- Issue that brought Visual Studio 2017 to crash
\ No newline at end of file
+- Issue that brought Visual Studio 2017 to crash.
+- Issuw with cache invalidation for Open Source vulnerabilities.
```
Comment (michelkaporin) [inline CHANGELOG.md:7]: Issue

**Review by michelkaporin — APPROVED** [review]

**Pull request merged**